packages/clerk-js/src/core/resources/UserSettings.ts

```diff
@@ -65,6 +67,7 @@ export class UserSettings extends BaseResource implements UserSettingsResource {
     this.attributes = Object.fromEntries(
       Object.entries(data.attributes).map(a => [a[0], { ...a[1], name: a[0] }]),
     ) as Attributes;
+    this.actions = data.actions;
```

Comment:
This small modification allows the environment context to have access to this new "actions" property which contains the values for permissions associated with user actions, which are needed for the user delete feature. Big thanks for @SokratisVidros for helping me to find this and patch this value into the right place.